core/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -454,6 +439,27 @@ public static GeoPoint parseGeoPoint(XContentParser parser, GeoPoint point) thro
         }
     }
 
+    public static GeoPoint parseNumericArray(XContentParser parser, GeoPoint point) throws IOException {
+        double lat = Double.NaN;
+        double lon = Double.NaN;
+        int element = 0;
+        while(parser.nextToken() != Token.END_ARRAY) {
+            if(parser.currentToken() == Token.VALUE_NUMBER) {
+                element++;
+                if(element == 1) {
+                    lon = parser.doubleValue();
+                } else if(element == 2) {
+                    lat = parser.doubleValue();
+                } else {
+                    throw new ElasticsearchParseException("only two values allowed");
```

Comment:
Can you add some context to this exception so the user knows what part of their request was wrong?